provider/tags_test.go

```diff
@@ -211,7 +214,123 @@ func TestApplyTags(t *testing.T) {
 			require.Equal(t, tc.expect, r)
 		})
 	}
 }
+
+func TestAddingEmptyTagProducesChangeDiff(t *testing.T) {
+	replayEvent := `
+	[
+	  {
+	    "method": "/pulumirpc.ResourceProvider/Diff",
+	    "request": {
+	      "id": "vpc-4b82e033",
+	      "urn": "urn:pulumi:testtags::tags-combinations-go::aws:ec2/defaultVpc:DefaultVpc::go-web-default-vpc",
+	      "olds": {
+		"__meta": "{\"schema_version\":\"1\"}",
+		"arn": "arn:aws:ec2:us-west-2:616138583583:vpc/vpc-4b82e033",
+		"assignGeneratedIpv6CidrBlock": false,
+		"cidrBlock": "172.31.0.0/16",
+		"defaultNetworkAclId": "acl-3b778d40",
+		"defaultRouteTableId": "rtb-a05f10db",
+		"defaultSecurityGroupId": "sg-4d436f12",
+		"dhcpOptionsId": "dopt-1649d26e",
+		"enableDnsHostnames": true,
+		"enableDnsSupport": true,
+		"enableNetworkAddressUsageMetrics": false,
+		"existingDefaultVpc": true,
+		"forceDestroy": false,
+		"id": "vpc-4b82e033",
+		"instanceTenancy": "default",
+		"ipv6AssociationId": "",
+		"ipv6CidrBlock": "",
+		"ipv6CidrBlockNetworkBorderGroup": "",
+		"ipv6IpamPoolId": "",
+		"ipv6NetmaskLength": 0,
+		"mainRouteTableId": "rtb-a05f10db",
+		"ownerId": "616138583583",
+		"tags": {
+		  "x": "s"
+		},
+		"tagsAll": {
+		  "x": "s"
+		}
+	      },
+	      "news": {
+		"__defaults": [
+		  "enableDnsHostnames",
+		  "enableDnsSupport",
+		  "forceDestroy"
+		],
+		"enableDnsHostnames": true,
+		"enableDnsSupport": true,
+		"forceDestroy": false,
+		"tags": {
+		  "__defaults": [],
+		  "x": "s",
+		  "y": ""
+		},
+		"tagsAll": {
+		  "__defaults": [],
+		  "x": "s",
+		  "y": ""
+		}
+	      },
+	      "oldInputs": {
+		"__defaults": [
+		  "enableDnsHostnames",
+		  "enableDnsSupport",
+		  "forceDestroy"
+		],
+		"enableDnsHostnames": true,
+		"enableDnsSupport": true,
+		"forceDestroy": false,
+		"tags": {
+		  "__defaults": [],
+		  "x": "s"
+		},
+		"tagsAll": {
+		  "__defaults": [],
+		  "x": "s"
+		}
+	      }
+	    },
+	    "response": {
+	      "changes": "DIFF_SOME",
+              "hasDetailedDiff": true
+	    }
+	  }
+	]`
+
+	replaySequence(t, replayEvent)
+}
+
+func TestTagsAllNoLongerComputed(t *testing.T) {
+	ctx := context.Background()
+	upstreamProvider, err := shim.NewUpstreamProvider(ctx)
+	require.NoError(t, err)
+
+	t.Run("sdk_v2", func(t *testing.T) {
+		for key, res := range upstreamProvider.SDKV2Provider.ResourcesMap {
+			tagsAll, ok := res.Schema["tags_all"]
+			if ok {
+				require.Falsef(t, tagsAll.Computed, "tags_all is Computed for %s", key)
+			}
+		}
+	})
+
+	// Note: when this test starts failing, see ./scripts/patch_computed_only.sh
+	t.Run("plugin_framework", func(t *testing.T) {
+		p := tfbridge.ShimProviderWithContext(ctx, upstreamProvider.PluginFrameworkProvider)
+		p.ResourcesMap().Range(func(key string, res tfshim.Resource) bool {
+			tagsAll, ok := res.Schema().GetOk("tags_all")
+			if ok {
+				require.Falsef(t, tagsAll.Computed(), "tags_all is Computed for %s", key)
+			}
+			return true
+		})
+	})
```

Comment:
Can we have a test for `Bucket`? It's one of the few resources that pulumi itself maintains and it has had tags bugs in the past.

Reply:
Yeah that's a very good catch. I'll add rn.

Reply:
Added. Passing locally 🤞